d1/main/newdemo.c

```diff
@@ -2172,7 +2172,7 @@ int newdemo_read_frame_information(int rewrite)
 				nd_write_short(distance);
 				break;
 			}
-			Players[Player_num].homing_object_dist = i2f((int)(distance << 16));
+			Players[Player_num].homing_object_dist = distance << 16;
```

Comment:
Is there going to be an overflow if `distance` isn't cast to a 32-bit data type before the bitshift? The old code would have suffered from this too, if I'm not mistaken about how C handles this...

Reply:
Double-checked... I am mistaken about it. The literal 16 is of integer type and will cause automatic type promotion before the bit-shift.